File: ios/Classes/ZendeskChannelMethod.swift
//
//  ZendeskChannelMethod.swift
//  mx_zendesk_messaging
//
//  Created by Magical Water on 2023/5/5.
//

import Foundation

enum ZendeskChannelMethod: String {
    case initialize = "initialize"
    case show = "show"
    case loginUser = "loginUser"
    case logoutUser = "logoutUser"
    case getUnreadMessageCount = "getUnreadMessageCount"
    case printLog = "printLog"
    
    // 事件觀察的未讀訊息數量變更
    case eventUnreadCount = "eventUnreadCount"
    
    // 事件觀察的調用api的驗證訊息錯誤
    case eventAuthFail = "eventAuthFail"
    
    // 事件觀察的未知錯誤
    case eventUnknown = "eventUnknown"
}

File: android/src/main/kotlin/org/water/mx_zendesk_messaging/ZendeskChannelMethod.kt
package org.water.mx_zendesk_messaging

enum class ZendeskChannelMethod(val rawValue: String) {
    Initialize("initialize"),
    Show("show"),
    LoginUser("loginUser"),
    LogoutUser("logoutUser"),
    GetUnreadMessageCount("getUnreadMessageCount"),
    PrintLog("printLog"),
    EventUnreadCount("eventUnreadCount"),
    EventAuthFail("eventAuthFail"),
    EventUnknown("eventUnknown");

    companion object {
        fun fromRawValue(rawValue: String): ZendeskChannelMethod? {
            return values().find { it.rawValue == rawValue }
        }
    }
}